interview-questions.md

```diff
@@ -9,43 +9,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is a hash in Ruby?
 
 Your answer:
+A hash in Ruby is like an array in JavaScript because it stores data in key-value pairs, but hashes don't use numbers as an index.
 
 Researched answer:
+Hashes can also be look at as dictionaries because they perform like them. The term iterate and its definition are an example of a hash:
+iterate - To execute a set of actions repeatedly to accomplish a task.
+In this example, the term "iterate" is the key and the definition that follows is the value. We access the data by looking up the word in the dictionary
 
 2. What is a gem?
 
 Your answer:
+A gem is Ruby code that has been made available for developers to use. Ruby on Rails is an example of a Ruby gem. Rails is a package of Ruby code that is made available to use in the development of an application. 
 
 Researched answer:
+Throughout our challenges over this past week of the cohort, we were using Ruby on Rails packages as we were creating models, working with schemas, and running migrations in our database manipulations. This particular gem provides the essential built-in code needed to connect to and work with backend portion of the application.
 
 3. What is Ruby on Rails?
 
 Your answer:
+Rails is the server-based framework that was written in Ruby and is used for backend web development and programming.
 
 Researched answer:
+Ruby on Rails is an MVC (Model, View, Controller) based framework and it gives a developer a different perspective of an application. Development in Rails is occurring on the backend side and it's data as opposed to the aesthetics of design that the front end of the application consists of. Rails is that glue that sticks the front end of an application to the back end of the application and makes for one complete package.
 
 4. What is a relational database? Are there other kinds of databases?
 
 Your answer:
+A relational database is the type of database that stores data in tables and gives access to that data through the use of related data points that correspond to the stored data.
 
 Researched answer:
+An active example of a relational database in use would be Amazon. The logisitics of Amazon's business thrives on using relational databases to complete orders by simply matching customer's information (customer_id, mailing address, email) to the actual order information (customer_id, item, cost, payment method) in order to complete a transaction. The customer's info and the order info could stored in different places, yet be connected to each other by a related key reference (customer_id) which brings the information all together to execute a task - sales and product distribution. There are 4 other kinds of databases besides the Relational database: Object-oriented, Network, Hierarchical, and NoSQL.
```

Comment:
good content here. More things you can share are brief definitions of the 4 other databases you mentioned.